ports/espressif/boards/m5stack_cardputer/board.c

```diff
@@ -0,0 +1,137 @@
+    // sw reset
+    0x01, 0 | DELAY, 150,
+    // sleep out
+    0x11, 0 | DELAY, 255,
+    // normal display mode on
+    0x13, 0,
+    // display and color format settings
+    0x36, 1, 0x68,
+    0xB6, 2, 0x0A, 0x82,
+    0x3A, 1 | DELAY,  0x55, 10,
+    // ST7789V frame rate setting
+    0xB2, 5, 0x0C, 0x0C, 0x00, 0x33, 0x33,
+    // voltages: VGH / VGL
+    0xB7, 1, 0x35,
+    // ST7789V power setting
+    0xBB, 1, 0x28,
+    0xC0, 1, 0x0C,
+    0xC2, 2, 0x01, 0xFF,
+    0xC3, 1, 0x10,
+    0xC4, 1, 0x20,
+    0xC6, 1, 0x0F,
+    0xD0, 2, 0xA4, 0xA1,
+    // ST7789V gamma setting
+    0xE0, 14, 0xD0, 0x00, 0x02, 0x07, 0x0A, 0x28, 0x32, 0x44, 0x42, 0x06, 0x0E, 0x12, 0x14, 0x17,
+    0xE1, 14, 0xD0, 0x00, 0x02, 0x07, 0x0A, 0x28, 0x31, 0x54, 0x47, 0x0E, 0x1C, 0x17, 0x1B, 0x1E,
+    0x21, 0,
+    // display on
+    0x29, 0 | DELAY, 255,
+};
+
+
+void board_init(void) {
+    busio_spi_obj_t *spi = common_hal_board_create_spi(0);
+    fourwire_fourwire_obj_t *bus = &allocate_display_bus()->fourwire_bus;
+    bus->base.type = &fourwire_fourwire_type;
+
+    // see here for inspiration: https://github.com/m5stack/M5GFX/blob/33d7d3135e816a86a008fae8ab3757938cee95d2/src/M5GFX.cpp#L1350
+    common_hal_fourwire_fourwire_construct(
+        bus,
+        spi,
+        &pin_GPIO34,    // DC
+        &pin_GPIO37,     // CS
+        &pin_GPIO33,    // RST
+        40000000,       // baudrate
+        0,              // polarity
+        0               // phase
+        );
+    busdisplay_busdisplay_obj_t *display = &allocate_display()->display;
+    display->base.type = &busdisplay_busdisplay_type;
+
+    common_hal_busdisplay_busdisplay_construct(
+        display,
+        bus,
+        240,            // width (after rotation)
+        135,            // height (after rotation)
+        40,             // column start
+        53,             // row start
+        0,              // rotation
+        16,             // color depth
+        false,          // grayscale
+        false,          // pixels in a byte share a row. Only valid for depths < 8
+        1,              // bytes per cell. Only valid for depths < 8
+        false,          // reverse_pixels_in_byte. Only valid for depths < 8
+        true,           // reverse_pixels_in_word
+        MIPI_COMMAND_SET_COLUMN_ADDRESS, // set column command
+        MIPI_COMMAND_SET_PAGE_ADDRESS,   // set row command
+        MIPI_COMMAND_WRITE_MEMORY_START, // write memory command
+        display_init_sequence,
+        sizeof(display_init_sequence),
+        &pin_GPIO38,    // backlight pin
+        NO_BRIGHTNESS_COMMAND,
+        1.0f,           // brightness
+        false,          // single_byte_bounds
+        false,          // data_as_commands
+        true,           // auto_refresh
+        60,             // native_frames_per_second
+        true,           // backlight_on_high
+        false,          // SH1107_addressing
+        50000           // backlight pwm frequency
```

Comment:
Cardputer display backlight is provided by the StampS3:

![image](https://github.com/adafruit/circuitpython/assets/22404700/0a63173f-87b5-4f29-a627-3ddfbb7b5598)

I did some test on the brightness and found out that with a PWM frequency of 50kHz any brightness value less then 1.0 make the screen completely dark. For example if the brightness is set to 0.9 the display goes completely black. I guess that this strange behavior is caused by the SGM2578

![image](https://github.com/adafruit/circuitpython/assets/22404700/6cee1c16-cbba-494b-8a50-b4faf841bd51)

With some experimentation I found out that display brightness works correctly with a PWM frequency to 350Hz.

```suggestion
        350             // backlight pwm frequency
```